modules/apps/static/portal-osgi-web/portal-osgi-web-portlet-tracker/src/main/java/com/liferay/portal/osgi/web/portlet/tracker/internal/PortletTracker.java

```diff
@@ -427,6 +429,69 @@ protected void collectCacheScope(
 		com.liferay.portal.kernel.model.Portlet portletModel) {
 	}
 
+	protected void collectContainerRuntimeOptions(
+		ServiceReference<Portlet> serviceReference,
+		com.liferay.portal.kernel.model.Portlet portletModel) {
+
+		PortletApp portletApp = portletModel.getPortletApp();
+
+		Map<String, String[]> containerRuntimeOptionsMap =
+			portletApp.getContainerRuntimeOptions();
+
+		List<String> containerRuntimeOptions = StringPlus.asList(
+			serviceReference.getProperty(
+				"javax.portlet.container-runtime-option"));
+
+		for (String containerRuntimeOption : containerRuntimeOptions) {
+			String name = containerRuntimeOption;
+			String[] parts = StringUtil.split(
+				containerRuntimeOption, CharPool.SEMICOLON);
+
+			if (parts.length == 2) {
+				name = parts[0];
+			}
+
+			containerRuntimeOptionsMap.remove(name);
+		}
+
+		for (String containerRuntimeOption : containerRuntimeOptions) {
+			String name = containerRuntimeOption;
+			String value = null;
+			String[] parts = StringUtil.split(
+				containerRuntimeOption, CharPool.SEMICOLON);
+
+			if (parts.length == 2) {
+				name = parts[0];
+				value = parts[1];
+			}
+
+			String[] values = containerRuntimeOptionsMap.get(name);
+
+			if (values == null) {
```

Comment:
@ngriffin7a can this "values" not be null? Did not we just remove the exact same name from the containerRuntimeOptionsMap in the first loop?
The later put here https://github.com/shuyangzhou/liferay-portal/pull/6408/commits/e1e9ab78e6de30320faecd7f39a44dd498531f47#diff-28e3cf259d89b86f287fb1efd5d889dfR490 is under a different name, on the next iteration it still won't see the "values" under "name", so the 2nd iteration would just throw away 1st iteration's "values", right?